docs/guides/mongodb/clustering/sharding.md

```diff
@@ -0,0 +1,1132 @@
+
+- To keep things isolated, this tutorial uses a separate namespace called `demo` throughout this tutorial. Run the following command to prepare your cluster for this tutorial:
+
+  ```console
+  $ kubectl create ns demo
+  namespace/demo created
+  ```
+
+> Note: The yaml files used in this tutorial are stored in [docs/examples/mongodb](https://github.com/kubedb/cli/tree/master/docs/examples/mongodb) folder in GitHub repository [kubedb/cli](https://github.com/kubedb/cli).
+
+## Deploy MongoDB Sharding
+
+To deploy a MongoDB Sharding, user have to specify `spec.replicaSet` option in `Mongodb` CRD.
+
+The following is an example of a `Mongodb` object which creates MongoDB Sharding of three members.
+
+```yaml
+apiVersion: kubedb.com/v1alpha1
+kind: MongoDB
+metadata:
+  name: mongo-sh
+  namespace: demo
+spec:
+  version: 3.6-v3
+  shardTopology:
+    configServer:
+      replicas: 3
+      storage:
+        resources:
+          requests:
+            storage: 1Gi
+        storageClassName: standard
+    mongos:
+      replicas: 2
+      strategy:
+        type: RollingUpdate
+    shard:
+      replicas: 3
+      shards: 3
+      storage:
+        resources:
+          requests:
+            storage: 1Gi
+        storageClassName: standard
+```
+
+```console
+$ kubectl create -f https://raw.githubusercontent.com/kubedb/cli/0.11.0//docs/examples/mongodb/clustering/mongo-sharding.yaml
+mongodb.kubedb.com/mongo-sh created
+```
+
+Here,
+
+- `spec.shardTopology` represents the topology configuration for sharding.
+  - `shard` represents configuration for Shard component of mongodb.
+    - `shards` represents number of shards for a mongodb deployment. Each shard is deployed as a [replicaset](/docs/guides/mongodb/clustering/replication_concept.md).
+    - `replicas` represents number of replicas of each shard replicaset.
+    - `prefix` represents the prefix of each shard node.
+    - `configSource` is an optional field to provide custom configuration file for shards (i.e mongod.cnf). If specified, this file will be used as configuration file otherwise default configuration file will be used.
+    - `podTemplate` is an optional configuration for pods.
+    - `storage` to specify pvcSpec for each node of sharding. You can specify any StorageClass available in your cluster with appropriate resource requests.
+  - `configServer` represents configuration for ConfigServer component of mongodb.
+    - `replicas` represents number of replicas for configServer replicaset. Here, configServer is deployed as a replicaset of mongodb.
+    - `prefix` represents the prefix of configServer nodes.
+    - `configSource` is an optional field to provide custom configuration file for configSource (i.e mongod.cnf). If specified, this file will be used as configuration file otherwise default configuration file will be used.
+    - `podTemplate` is an optional configuration for pods.
```

Comment:
We need to point to a doc that explains the options available to a `podTemplate`.